pipelines/template-build/template-build.yaml

```diff
@@ -1,394 +1,410 @@
-# This build pipeline template is intended to be processed by scripts
-# under hack/ directory rather than by kustomize directly.
-
 apiVersion: tekton.dev/v1
 kind: Pipeline
 metadata:
-  name: template-build
   labels:
-    "pipelines.openshift.io/used-by": ""
-    "pipelines.openshift.io/runtime": ""
-    "pipelines.openshift.io/strategy": ""
+    pipelines.openshift.io/runtime: ""
+    pipelines.openshift.io/strategy: ""
+    pipelines.openshift.io/used-by: ""
+  name: template-build
 spec:
-  params:
-    - description: 'Source Repository URL'
-      name: git-url
-      type: string
```

Comment:
Oof, I didn't realize this would make massive formatting + ordering changes to files that didn't need any changes.

I'm trying to think of a way to avoid that. If we didn't need the kustomization files that just reference the yaml by itself, that would be great (e.g. https://github.com/konflux-ci/build-definitions/blob/a57eb69abd22232e19219b977a89e54386292a3f/pipelines/template-build/kustomization.yaml#L1-L4

Reply:
Maybe we can just exclude that in the script? If the kustomization file only has one element in `resources`, and that element is the task/pipeline yaml, skip it?

Reply:
In this PR, the `formatting + ordering` changes (where we have single `resources` in kustomization.yaml) are of one time change, do you anticipate any issue if we merge this change? one issue I can think of is we might need to rework/rebase if any open PR is changing those files.
If we decide not to do the formatting/ordering, it is possible to exclude in the script.

Reply:
They are a one-time change now, but this will cause PRs to often fail CI checks due to irrelevant formatting changes in the future. Which could be acceptable if the formatting made sense, but I don't like the object key sorting. It forces things into an unnatural reading order, where e.g. the `default` for a parameter comes before its `name`

Reply:
And, pragmatically, excluding them will make this PR require _much_ fewer approvals :grin:

Reply:
@chmeliik Excluded those task/pipelines programmatically now, please take a look agian.